src/include/simeng/arch/aarch64/helpers/neon.hh

```diff
@@ -785,14 +785,21 @@ RegisterValue vecSumElems_2ops(srcValContainer& sourceValues) {
 template <typename D, typename N, int I>
 RegisterValue vecXtn(srcValContainer& sourceValues, bool isXtn2) {
   const D* d;
-  if (isXtn2) d = sourceValues[0].getAsVector<D>();
-  const N* n = sourceValues[isXtn2 ? 1 : 0].getAsVector<N>();
+  const N* n;
+  if (isXtn2) {
+    d = sourceValues[0].getAsVector<D>();
+    n = sourceValues[1].getAsVector<N>();
+  } else {
+    d = {};
+    n = sourceValues[0].getAsVector<N>();
+  }
 
   D out[16 / sizeof(D)] = {0};
   int index = 0;
 
   for (int i = 0; i < I; i++) {
     if (isXtn2 & (i < (I / 2))) {
+      assert(isXtn2 && "isXtn2 is false so d is not initialised");
       out[i] = d[i];
     } else {
       out[i] = static_cast<D>(n[index]);
```

Comment:
@FinnWilkinson this is a temporary solution as I didn't want to fiddle too much with the for loop below. If you have a preferred solution then feel free to push a commit implementing that.

Reply:
I think this proposed solution is good. 

From our offline discussion aboout line 801 on whether it should be `isXtn2 & (i < (I / 2))` or `isXtn2 && (i < (I / 2))`, given both arguments are bools the functionality would be the same, but using `&&` may be marginally clearer of this fact?